openshift-hack/commitchecker/validate.go

```diff
@@ -19,6 +20,46 @@ var (
 	}
 )
 
+// ValidateReleaseBranchConsistency outputs an error in case we're missing a commit in
+// any of the following (n+2) release branches. This helps us to understand if we're merging a fix that isn't present
+// in later versions of OCP yet - which might cause regression problems in cluster upgrades.
+// This currently only works on release-4.x branches for simplicity.
+func ValidateReleaseBranchConsistency(commits []Commit) []string {
+	var allErrors []string
+
+	branch, err := CurrentBranch()
+	if err != nil {
+		return append(allErrors, fmt.Sprintf("Could not retrieve branch information: %v", err))
+	}
+
+	if IsReleaseBranch(branch) {
+		version, err := ReleaseBranchYVersion(branch)
+		if err != nil {
+			return append(allErrors, fmt.Sprintf("Could not retrieve y version information: %v", err))
+		}
+
+		releaseBranches := []string{
+			fmt.Sprintf("release-4.%d", version+1),
+			fmt.Sprintf("release-4.%d", version+2),
+		}
+
+		for _, branch := range releaseBranches {
+			err := FetchBranch(branch)
+			if err == nil {
+				for _, commit := range commits {
+					if IsCommitMissingInBranch(commit, branch) {
```

Comment:
I'm a bit hesitant about commit matching, this might have too many false positives, imo.

Reply:
Have you checked how many false positives we currently have on our branches?

Reply:
plenty, you can check with `git log --cherry-mark --no-merges` and between the releases. **But**,  this tool works only on the PR change set, so you will only ever get the commits back that you added within the PR (worst case) and that are missing in release+N.

On rebases this might be large, but I reckon it aids a bit in looking at whether there is something important missing. 
I'm a bit more worried about the day2day normal carry cherry-picks and build errors, but they should in theory be done in that order anyway.

Reply:
hmm, but even looking into the other picks, the commit sha's can be entirely different:
https://github.com/openshift/kubernetes/pull/1056 => 1bc6c1c
https://github.com/openshift/kubernetes/pull/1055 => f7d38b0
https://github.com/openshift/kubernetes/pull/1054 => 4472d10

it won't actually work that way- is there any common ground we can try to match on? the git commit message is likely also not helpful

Reply:
@tjungblu here's my proposal, let add it here, but hide it behind a flag. Then we can add a separate CI check which will return :+1: most of the time, unless it finds merge commit, that's pretty easy to find. We can make that job an optional for starters and let's see how well it will handle the next few bumps and then we can make the decision about further actions.

Reply:
sounds great @soltysh, I'll prep the whole thing and will let you know next week.

Reply:
@soltysh finally found time. I put the whole thing behind a flag now "--check-rebase". Will add another check in the release repo once this is in. PTAL :)